### PR TITLE
Kubectl config

### PR DESCRIPTION
**Pull request by mbruzek**
Adding the logic to generate a kubectl package for someone to download and use kubectl to directly control Kubernetes.

### CODE REVIEW

File: reactive/k8s.py [hunk 1]
```diff
@@ -31,6 +31,11 @@ def config_changed():
         docker_compose_kill_remove('proxy')
         remove_state('proxy.available')
 
+        if config.changed('version'):
+            hookenv.log('Removing kubectl.downloaded state so the new version'
+                        'kubectl will be downloaded again.')
```
Comment (lazypower) [inline reactive/k8s.py]: add "of' or omit kubectl

File: reactive/k8s.py [hunk 2]
```diff
@@ -80,6 +85,33 @@ def download_kubectl():
     set_state('kubectl.downloaded')
     status_set('active', 'Kubernetes installed')
 
+
+@when('kubectl.downloaded')
+@when_not('kubectl.package.created')
+def package_kubectl():
+    '''Package the kubectl binary and configuration to a tar file for users
+    to consume and interact directly with Kubernetes.'''
+    cluster_name = 'kubernetes'
```
Comment (lazypower) [inline reactive/k8s.py:94]: Are we hard coding the cluster name? It wound up not being required for the DNS bits. we can expose this as a config option I think.

File: reactive/k8s.py [hunk 3]
```diff
@@ -80,6 +85,33 @@ def download_kubectl():
...
+          ' --cluster={0}'
+    check_call(split(cmd.format(cluster_name)))
+    # TODO: Set the ca, cert and users via kubecfg
+    # Copy the kubectl to /tmp/
+    cmd = 'cp -v /usr/local/bin/kubectl /tmp/'
+    check_call(split(cmd))
+    # Zip that file up.
+    with chdir('/tmp'):
+        cmd = 'tar -cvzf ../kubectl_package.tar.gz kubectl .kube'
+        check_call(split(cmd))
```
Comment (lazypower) [inline reactive/k8s.py:112]: :+1: I like this approach as a one size fits all

File: reactive/k8s.py [hunk 4]
```diff
@@ -89,6 +121,7 @@ def start_cadvisor():
     status_set('active', 'cadvisor running on port 8088')
     hookenv.open_port(8088)
 
+
```
Comment (lazypower) [inline reactive/k8s.py:124]: DEATH TO PEP8 NEWLINES! :bomb:

File: reactive/k8s.py (hunk 1 above)
Comment (mbruzek) [inline reactive/k8s.py]: Good catch, thanks!

**Comment by lazypower** [comment]
Minor comments, but overall LGTM - pulling and testing now

File: reactive/k8s.py (hunk 2 above)
Comment (mbruzek) [inline reactive/k8s.py:94]: It is currently "kubernetes" in the master.json config file.  We do not have this configurable.  Not sure it needs to be configurable but we could if a use case make that useful.

**Pull request merged**